### DISCUSSION TITLE
File Nesting a la JetBrains

### DISCUSSION THREAD

**Discussion by bennypowers** [nvim-neo-tree/neo-tree.nvim#117] [Ideas]
Jetbrains (webstorm, etc) has a feature where file entries in the tree can collapse into one another. For example, a user can configure the tree to show `*.ts` files as a subtree, with siblings matching `$1.d.ts` `$1.js` and `$1.js.map` as the subtree's children:

![example of nesting](https://resources.jetbrains.com/help/img/idea/2021.3/nesting-rules-on-off.png)

This would be a cool feature in neo-tree. I'm wondering if maintainers and contributors think this could be easily accomplished in userland?

**Comment by cseickel** [comment]
That is really cool! You can probably hack it in in an `after_render` event, but this is best done as a change to the internal code. I'm going to create an issue for this and put it on the to-do list.

**Reply by bennypowers** [reply]
here's some jb docs on the configuration

https://www.jetbrains.com/help/rider/File_Nesting_Dialog.html

I'd imagine something like

```lua
file_nesting = {
  "ts": { "cjs", "cjs.map", "js", "js.map", "d.ts" },
},
```
